parsing.go

```diff
@@ -283,21 +314,164 @@ func parseReq(txt string) (*Req, error) {
 		return nil, fmt.Errorf("Requirement body must not be empty: %s", r.ID)
 	}
 
+	// PARENTS must be punctuation/space separated list of parseable req-ids.
+	err = parseParents(r)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// parseReqTable reads a table of requirements one row at a time and parses the content into Req structures which are
+// then returned in a slice.
+//
+// Tables have the following format:
+// | ID | Title | Body | Attribute1 | Attribute2 |
+// | --- | --- | --- | --- | --- |
+// | ReqID | <text> | <text> | <text> | <text> |
+//
+// The first column must be "ID" and each row must contain a valid ReqID. Other columns are optional.
+//
+// @llr REQ-TRAQ-SWL-25
+func parseReqTable(txt string, reqs []*Req) ([]*Req, error) {
+
+	var attributes []string
+
+	// Split the table into rows and loop through
+	for rowN, rowS := range strings.Split(txt, "\n") {
+
+		// The first row contains the attribute names for each column, the first column must be "ID"
+		if rowN == 0 {
+			if reTableHeader.MatchString(rowS) {
+				attributes = splitTableLine(rowS)
+				for i, a := range attributes {
+					attributes[i] = strings.ToUpper(a)
+				}
+			} else {
+				return reqs, fmt.Errorf("first column must be \"ID\"")
+			}
+		} else {
+			if reTableDelimiter.MatchString(rowS) {
+				// Ignore the delimiter row
+				continue
+			}
+
+			values := splitTableLine(rowS)
+
+			if len(values) == 0 {
+				// End of table
+				break
+			}
+
+			if len(values) < len(attributes) {
+				return reqs, fmt.Errorf("too few cells on row %d of requirement table", rowN+1)
+			}
+
+			r := &Req{Attributes: map[string]string{}}
+
+			// For each attribute in the first row, read in the associated value on this row
+			for i, k := range attributes {
+				if k == "ID" {
+					ID, IDNumber, err := extractIDAndNum(values[i])
+					if err != nil {
+						return reqs, err
+					}
+					r.ID = ID
+					r.IDNumber = IDNumber
+				} else if k == "TITLE" {
+					r.Title = values[i]
+				} else if k == "BODY" {
+					r.Body = values[i]
+				} else {
+					if k == "PARENT" {
```

Comment:
instead of doing this check for every row, do it once in the rowN == 0 block above.